dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/wrapper/ScopeClusterInvoker.java

```diff
@@ -0,0 +1,213 @@
+
+    private boolean peerFlag;
+
+    private boolean injvmFlag;
+
+    private final Object createLock = new Object();
+
+
+    public ScopeClusterInvoker(Directory<T> directory, Invoker<T> invoker) {
+        this.directory = directory;
+        this.invoker = invoker;
+        this.isExported = new AtomicBoolean(false);
+        init();
+    }
+
+    private void init() {
+        Boolean peer = (Boolean) getUrl().getAttribute(PEER_KEY);
+        String isInjvm = getUrl().getParameter(LOCAL_PROTOCOL);
+        if (peer != null && peer) {
+            peerFlag = true;
+            return;
+        }
+        if (injvmInvoker == null && LOCAL_PROTOCOL.equalsIgnoreCase(getRegistryUrl().getProtocol())) {
+            injvmInvoker = invoker;
+            isExported.compareAndSet(false, true);
+            injvmFlag = true;
+            return;
+        }
+        if (Boolean.TRUE.toString().equalsIgnoreCase(isInjvm) || LOCAL_KEY.equalsIgnoreCase(getUrl().getParameter(SCOPE_KEY))) {
+            injvmFlag = true;
+        } else if (isInjvm == null) {
+            injvmFlag = isNotRemoteOrGeneric();
+        }
+
+        protocolSPI = getUrl().getApplicationModel().getExtensionLoader(Protocol.class).getAdaptiveExtension();
+        injvmExporterListener = getUrl().getOrDefaultFrameworkModel().getBeanFactory().getBean(InjvmExporterListener.class);
+        injvmExporterListener.addExporterChangeListener(this, getUrl().getServiceKey());
+    }
+
+    @Override
+    public URL getUrl() {
+        return directory.getConsumerUrl();
+    }
+
+    @Override
+    public URL getRegistryUrl() {
+        return directory.getUrl();
+    }
+
+    @Override
+    public Directory<T> getDirectory() {
+        return directory;
+    }
+
+    @Override
+    public boolean isDestroyed() {
+        return directory.isDestroyed();
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return isExported.get() || directory.isAvailable();
+    }
+
+    @Override
+    public void destroy() {
+        if (injvmExporterListener != null) {
+            injvmExporterListener.removeExporterChangeListener(this, getUrl().getServiceKey());
+        }
+        destroyInjvmInvoker();
+        this.invoker.destroy();
+    }
+
+    @Override
+    public Class<T> getInterface() {
+        return directory.getInterface();
+    }
+
+    @Override
+    public Result invoke(Invocation invocation) throws RpcException {
+        if (peerFlag) {
+            return invoker.invoke(invocation);
+        }
+        if (isInjvmExported() && injvmFlag) {
+            return injvmInvoker.invoke(invocation);
+        }
+        return invoker.invoke(invocation);
+    }
+
+    private boolean isNotRemoteOrGeneric() {
+        return !SCOPE_REMOTE.equalsIgnoreCase(getUrl().getParameter(SCOPE_KEY)) &&
+            !getUrl().getParameter(GENERIC_KEY, false);
+    }
+
+    private boolean isInjvmExported() {
+        Boolean localInvoke = RpcContext.getServiceContext().getLocalInvoke();
```

Comment:
`RpcContext.getServiceContext().getLocalInvoke()` here should only effect **one** invocation.
If `null`, ignore.
If `false`, force remote.
If `true`, force local.

Reply:
done